Fix date pattern detection/double-quoted strings

MalteWunsch opened the pull request:
Excel uses double-quoted strings to indicate raw text. So when looking for date format patterns in custom cell formats, we should ignore double-quoted strings to prevent false-positives.

MalteWunsch committed:
Fix date pattern detection/double-quoted strings

Excel uses double-quoted strings to indicate raw text. So when looking for date format patterns in custom cell formats, we should ignore double-quoted strings to prevent false-positives.

MalteWunsch committed:
Apply php-cs-fixer

MalteWunsch commented:
@Slamdunk The php-cs-fixer fails are unrelated - should I include their fixes anyway?

Slamdunk commented:
Yes please

MalteWunsch committed:
Apply php-cs-fixer to unrelated files

MalteWunsch commented:
Included ✅

Slamdunk commented:
Can you give me an Excel with such data, so I can do a manual test before merging please?

MalteWunsch commented:
@Slamdunk Here you are: [test.xlsx](https://github.com/openspout/openspout/files/12317223/test.xlsx)

MalteWunsch committed:
Improve doc blocks

These files are unrelated to this PR, but the missing doc blocks let phpstan fail.
